Core/src/Surfaces/LineSurface.cpp

```diff
@@ -80,27 +79,20 @@ Acts::Result<Acts::Vector2> Acts::LineSurface::globalToLocal(
     const GeometryContext& gctx, const Vector3& position,
     const Vector3& direction, double tolerance) const {
   using VectorHelpers::perp;
-  const auto& sTransform = transform(gctx);
-  const auto& tMatrix = sTransform.matrix();
-  Vector3 lineDirection(tMatrix(0, 2), tMatrix(1, 2), tMatrix(2, 2));
-  // Bring the global position into the local frame
-  Vector3 loc3Dframe = sTransform.inverse() * position;
-  // construct localPosition with sign*perp(candidate) and z.()
-  Vector2 lposition(perp(loc3Dframe), loc3Dframe.z());
-  Vector3 sCenter(tMatrix(0, 3), tMatrix(1, 3), tMatrix(2, 3));
-  Vector3 decVec(position - sCenter);
-  // assign the right sign
-  double sign = ((lineDirection.cross(direction)).dot(decVec) < 0.) ? -1. : 1.;
-  lposition[eBoundLoc0] *= sign;
-
-  // TODO make intersection and local<->global tolerances sound. quick fixed
-  // this with a 50% tolerance increase for now
-  if ((localToGlobal(gctx, lposition, direction) - position).norm() >
-      tolerance * 1.5) {
+
+  // Bring the global position into the local frame. First remove the
+  // translation then the rotation.
+  Vector3 localPosition = referenceFrame(gctx, position, direction).inverse() *
+                          (position - transform(gctx).translation());
```

Comment:
Ok this is very clear now, thanks! Did you check how the output of the function changes with the new code?

Reply:
this is validated through physmon / unit tests I'd say. the point is that the output changes tho